pkg/conversion/mapi2capi/aws_test.go

```diff
@@ -16,43 +16,140 @@ limitations under the License.
+	Entry("With mismatched region", awsMAPIMachineWithMismatchedRegion.Build(), infraWithRegion.Build(), MatchError(ContainSubstring(errPlacementRegionShouldMatchInfrastructureStatusValue)), BeEmpty()),
+	Entry("With metadata in provider spec", awsMAPIMachineWithMetadata.Build(), infra, MatchError(ContainSubstring(errMetadataNotSupported)), BeEmpty()),
+	Entry("With unsupported network interface type", awsMAPIMachineWithInvalidNIC.Build(), infra, MatchError(ContainSubstring(errNetworkInterfaceUnsupportedValue)), BeEmpty()),
+	Entry("With AMI ARN reference", awsMAPIMachineWithAMIARN.Build(), infra, MatchError(ContainSubstring(errUnableToConvertUnsupportedAMIARNReference)), BeEmpty()),
+	Entry("With AMI filters", awsMAPIMachineWithAMIFilter.Build(), infra, MatchError(ContainSubstring(errUnableToConvertUnsupportedAMIFilterReference)), BeEmpty()),
+	Entry("With missing AMI reference", awsMAPIMachineWithMissingAMI.Build(), infra, MatchError(ContainSubstring(errUnableToFindAMIReference)), BeEmpty()),
+	Entry("With unsupported Metadata Authentication", awsMAPIMachineWithInvalidMetadata.Build(), infra, MatchError(ContainSubstring(errUnsupportedAuthenticationValue)), BeEmpty()),
+	Entry("With missing Volume size for EBS", awsMAPIMachineWithMissingVolumeSize.Build(), infra, MatchError(ContainSubstring(errVolumeSizeRequiredButMissing)), BeEmpty()),
+	Entry("With non-root Volume not deleted on termination", awsMAPIMachineWithNonRootVolumeNotDeleted.Build(), infra, MatchError(ContainSubstring(errNonRootVolumesMustBeDeletedOnTerminationUnsupportedValueFalse)), BeEmpty()),
+	Entry("With NoDevice specified", awsMAPIMachineWithNoDevice.Build(), infra, MatchError(ContainSubstring(errNoDeviceIsNotSupported)), BeEmpty()),
+	Entry("With VirtualName specified", awsMAPIMachineWithVirtualName.Build(), infra, MatchError(ContainSubstring(errVirtualNameNotSupported)), BeEmpty()),
+	// Error + Warning.
+	Entry("With VirtualName specified and missing EBS configuration", awsMAPIMachineWithVirtualNameAndMissingEBSConfig.Build(), infra, MatchError(ContainSubstring(errVirtualNameNotSupported)), ConsistOf(ContainSubstring(warnMissingBlockDeviceConf))),
+	Entry("With VirtualName specified and root Volume not deleted on termination", awsMAPIMachineWithVirtualNameAndRootVolNotDeletedOnTerm.Build(), infra, MatchError(ContainSubstring(errVirtualNameNotSupported)), ConsistOf(ContainSubstring(warnRootVolumeMustBeDeletedOnTerminationIgnoringValueFalse))),
+	// Double Errors.
+	Entry("With NoDevice and VirtualName specified", awsMAPIMachineWithNoDeviceAndVirtualName.Build(), infra, ConsistOf(MatchError(ContainSubstring(errNoDeviceIsNotSupported)), MatchError(ContainSubstring(errVirtualNameNotSupported))), BeEmpty()),
+	// Only Warnings.
+	Entry("With missing EBS configuration", awsMAPIMachineWithMissingEBSConfig.Build(), infra, BeNil(), ConsistOf(ContainSubstring(warnMissingBlockDeviceConf))),
```

Comment:
You call `Build()` on every entry for the Machine, have you considered passing a builder into the function and calling build on it within the describe table function?